pkg/river/internal/value/decode.go

```diff
@@ -147,6 +146,62 @@ func decode(val Value, into reflect.Value) error {
 	}
 }
 
+func decodeAny(val Value, into reflect.Value) error {
```

Comment:
There needs to be more doc/clarification around the differences between `decodeAny` and `decode`. That they can recursively call into each other is confusing. They have very similar names and same args? I *think* decodeAny is the backdoor for when we dont concretely know the type and its an interface{}?

Reply:
Right, `decodeAny` is when we're decoding into a `interface{}` and we want to provide a set of fixed types based on the River value. We recursively call `decode` with the non-`interface{}` type to decode into. 

I'll try to update the docs here to make it more clear that's what's going on.